numpy/core/tests/test_multiarray.py

```diff
@@ -8513,9 +8513,31 @@ def test_1d_no_format(self):
         assert_equal('{}'.format(a), str(a))
 
     def test_1d_format(self):
-        # until gh-5543, ensure that the behaviour matches what it used to be
-        a = np.array([np.pi])
-        assert_raises(TypeError, '{:30}'.format, a)
+        # gh-5543
+        a = np.array([-np.pi, np.pi])
+        fmtspc = "+.4f"
+        assert_equal(format(a, fmtspc), np.array_format(a, fmtspc))
```

Comment:
Could you extend the tests to all the built-in dtypes?  I am curious how this deals with datetime, int, float128, complex...

Reply:
Just did in my last commit